From cac77b6674b479788cd688f1aee62a764cfbbb41 Mon Sep 17 00:00:00 2001
From: Niklas Haas <git@haasn.dev>
Date: Fri, 8 Jul 2022 22:41:10 +0200
Subject: [PATCH] glsl: glslang back compat

This include path randomly changed at some point in time.
---
 src/glsl/glslang.cc | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/glsl/glslang.cc b/src/glsl/glslang.cc
index ba16458d..a9b34f64 100644
--- a/src/glsl/glslang.cc
+++ b/src/glsl/glslang.cc
@@ -26,15 +26,20 @@ extern "C" {
 
 #include <glslang/Include/ResourceLimits.h>
 #include <glslang/Public/ShaderLang.h>
-#include <glslang/SPIRV/GlslangToSpv.h>
-
-#include "glslang.h"
 
 #define GLSLANG_VERSION_CHECK(major, minor, patch) \
     (((major) < GLSLANG_VERSION_MAJOR) || ((major) == GLSLANG_VERSION_MAJOR && \
     (((minor) < GLSLANG_VERSION_MINOR) || ((minor) == GLSLANG_VERSION_MINOR && \
      ((patch) <= GLSLANG_VERSION_PATCH)))))
 
+#if GLSLANG_VERSION_CHECK(0, 0, 3496)
+#include <glslang/SPIRV/GlslangToSpv.h>
+#else
+#include <SPIRV/GlslangToSpv.h>
+#endif
+
+#include "glslang.h"
+
 using namespace glslang;
 
 static pthread_mutex_t pl_glslang_mutex = PTHREAD_MUTEX_INITIALIZER;
-- 
2.36.1

